.grit/patterns/js/replace-zlib-deflate-to-zlib-deflateSync.md

```diff
@@ -0,0 +1,94 @@
+---
+title: Replace `zlib.deflate` ⇒ `zlib.deflateSync`
+---
+
+Creating and using a large number of `zlib` objects simultaneously can cause significant memory fragmentation. It is strongly recommended that the results of compression operations be cached or made synchronous to avoid duplication of effort
+
+- [reference](https://nodejs.org/api/zlib.html#zlib_threadpool_usage_and_performance_considerations)
+
+tags: #fix #best-practice
+
+```grit
+engine marzano(0.1)
+language js
+
+or {
+    `for($a;$b;$c){$body}`,
+    `$arr.forEach(($condition) => {$body})`,
+    `while($condition){$body}`,
+    `do {$body} while($condition)`
```

Comment:
We should extract these into a "loop_like" pattern, similar to https://github.com/getgrit/stdlib/blob/870749f0ef23b2176f81d25af2d95bcdc62c7659/.grit/patterns/js/marzano.grit#L8

Reply:
Hi @morgante, updated as a function